testgrid/conformance/upload_e2e.py

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env python
+
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script parses conformance test output to produce testgrid entries
+#
+# Assumptions:
+# - there is one log file and one JUnit file (true for current conformance tests..)
+# - the log file contains ginkgo's output (true for kubetest and sonobuoy..)
+# - the ginkgo output will give us start / end time, and overall success
+#
+# - the start timestamp is suitable as a testgrid ID (unique, monotonic)
+#
+# - the test ran in the current year unless --year is provided
+# - the timestamps are parsed on a machine with the same local time (zone) 
+#   settings as the machine that produced the logs
+#
+# The log file is the source of truth for metadata, the JUnit will be consumed
+# by testgrid / gubernator for individual test case results
+#
+# Usage: see README.md
+
+
+import re
+import sys
+import time
+from datetime import datetime
+import argparse
+import json
+import subprocess
+from os import path
+
+# NOTE e2e logs use go's time.StampMilli ("Jan _2 15:04:05.000")
+# Example log line with a timestamp:
+# Jan 26 06:38:46.284: INFO: Running AfterSuite actions on all node
+# the third ':' separates the date from the rest
+E2E_LOG_TIMESTAMP_RE = re.compile('(... .\d \d\d:\d\d:\d\d\.\d\d\d):.*')
```

Comment:
Use `r'\d'` to make sure `\` is not part of an escape (similar to backtick in golang)

Reply:
done.